src/test/DeltaReturningHook.sol

```diff
@@ -36,20 +38,28 @@ contract DeltaReturningHook is BaseTestHooks {
         deltaSpecified = delta;
     }
 
-    function setDeltaUnspecified(int128 delta) external {
-        deltaUnspecified = delta;
+    function setDeltaUnspecifiedBeforeSwap(int128 delta) external {
+        deltaUnspecifiedBeforeSwap = delta;
+    }
+
+    function setDeltaUnspecifiedAfterSwap(int128 delta) external {
+        deltaUnspecifiedAfterSwap = delta;
     }
 
     function beforeSwap(
         address, /* sender **/
         PoolKey calldata key,
         IPoolManager.SwapParams calldata params,
         bytes calldata /* hookData **/
-    ) external override onlyPoolManager returns (bytes4, int128) {
-        (Currency specifiedCurrency,) = _sortCurrencies(key, params);
-        _settleOrTake(specifiedCurrency, deltaSpecified);
+    ) external override onlyPoolManager returns (bytes4, BeforeSwapDelta) {
+        (Currency specifiedCurrency, Currency unspecifiedCurrency) = _sortCurrencies(key, params);
+
+        if (deltaSpecified != 0) _settleOrTake(specifiedCurrency, deltaSpecified);
+        if (deltaUnspecifiedBeforeSwap != 0) _settleOrTake(unspecifiedCurrency, deltaUnspecifiedBeforeSwap);
```

Comment:
Would it be more optimal to settle deltaUnspecified always afterSwap if deltaUnspecifiedAfterSwap > 0 ? Otherwise you have to settle/take twice and they may net.

Reply:
Currently kept this hook like this so that our tests can just enable 1 hook or the other for flexibility